src/components/diff/model.ts

```diff
@@ -6,10 +6,11 @@ import { Git } from '../../tokens';
  * Base DiffModel class
  */
 export class DiffModel<T> implements IDisposable, Git.Diff.IModel<T> {
```

Comment:
Could you try removing the template on `DiffModel`? Now that we figure out how to deal with different type directly in the merge widget, it should be fine.

Reply:
Doesn't this also mean I need to remove the template from `Diff.IModel` and `Diff.IContent`as well?
Or do we want to hard code `string` type?
```suggestion
export class DiffModel implements IDisposable, Git.Diff.IModel<string> {
```

Reply:
Yes please remove the template as there is no other case than file content being of type string.